Allow for Extracted Word Cloud Block | Fix Test Cases

farhan opened the pull request:
In this PR:
- We have added functionality to run Word Cloud XBlock test cases for Built-In and Extracted XBlocks.
- We have fixed the test cases for Extracted Word Cloud XBlock.

~In this PR we have enabled the usage of Extracted Word Cloud XBlock resides in `xblocks-contrib` repository.~

feanil commented on lms/djangoapps/courseware/tests/test_word_cloud.py:
```diff
@@ -223,10 +226,15 @@ def test_word_cloud_constructor(self):
 
-        assert fragment.content == self.runtime.render_template('word_cloud.html', expected_context)
+        if settings.USE_EXTRACTED_WORD_CLOUD_BLOCK:
+            expected_context['range_num_inputs'] = range(5)
```
Why not check the block type in this case?

farhan replied:
Fixed, checking for both (Builtin & Extracted) XBlocks now.

feanil commented on xmodule/tests/test_word_cloud.py:
```diff
@@ -42,7 +45,14 @@ def test_xml_import_export_cycle(self):
+        if settings.USE_EXTRACTED_WORD_CLOUD_BLOCK:
+            def_id = runtime.id_generator.create_definition(olx_element.tag, olx_element.get('url_name'))
+            keys = ScopeIds(None, olx_element.tag, def_id, runtime.id_generator.create_usage(def_id))
+            block = WordCloudBlock.parse_xml(olx_element, runtime, keys)
```
Why do keys need to be provided for one instantiation but not the other?

farhan replied:
**Update:**
I have added keys for both builtin and extracted implementation to avoid If condition.

**Why It's not needed in case of builtin xblock implementation?**
In BuiltIn xblock implementation `WordCloud` block has [XmlMixin](https://github.com/openedx/edx-platform/blob/919cc78be1004140b4ce9e4ac93acde487e81625/xmodule/xml_block.py#L103) and in its [parse_xml](https://github.com/openedx/edx-platform/blob/919cc78be1004140b4ce9e4ac93acde487e81625/xmodule/xml_block.py#L292) method implementation, there is a check at start that if keys are None then it creates the keys.
I have copied the same lines of code and added them in test case.
In Extracted XBlock implementation [parse_xml](https://github.com/openedx/XBlock/blob/102ff5c208eb43d1aa77adf42dd5de25916873eb/xblock/core.py#L736-L737) method of `xblocks/core.py` calls which doesn't have this check and expecting keys values in params.

farhan replied:
Further discussion:
https://github.com/openedx/edx-platform/pull/35983#discussion_r1928767464

feanil commented on xmodule/tests/test_word_cloud.py:
```diff
@@ -53,12 +63,26 @@ def test_xml_import_export_cycle(self):
-        # This will export the olx to a separate file.
-        block.add_xml_to_node(node)
+        if settings.USE_EXTRACTED_WORD_CLOUD_BLOCK:
+            filepath = 'word_cloud/block_id.xml'
```
Why not use the same method as below to export the file?

farhan replied:
Because [add_xml_to_node](https://github.com/openedx/edx-platform/blob/919cc78be1004140b4ce9e4ac93acde487e81625/xmodule/xml_block.py#L426) is defined in [XmlMixin](https://github.com/openedx/edx-platform/blob/919cc78be1004140b4ce9e4ac93acde487e81625/xmodule/xml_block.py#L103) class.
BuiltIn XBlocks are using this mixin

In Extracted XBlocks we are preferring to use Xblock.core functionalities and using [export_to_xml](https://github.com/openedx/XBlock/blob/5b91160342e563f7829d9ae22a6d15bb42d015a2/xblock/runtime.py#L773) method which doesn't have `export_to_file` funcationlity like `add_xml_to_node` method of `XmlMixin` has

feanil reviewed (commented):
I took a pass but since I'm not as familiar with the Xblock framework, forgive me if some of the questions have obvious answers.  I'm in particular confused about why the tests need to change and if that indicates any sort of breakage that we need to handle elsewhere?

farhan commented:
> I took a pass but since I'm not as familiar with the Xblock framework, forgive me if some of the questions have obvious answers. I'm in particular confused about why the tests need to change and if that indicates any sort of breakage that we need to handle elsewhere?

@feanil Thanks for the pass.
Let me review my implementation, rethink on them, and address your rightful queries.

Let's keep this PR in draft until I make this PR and Extracted Word Cloud PR https://github.com/openedx/xblocks-contrib/pull/4 fully ready.

farhan commented on xmodule/tests/test_word_cloud.py:
```diff
@@ -42,7 +45,11 @@ def test_xml_import_export_cycle(self):
+
+        def_id = runtime.id_generator.create_definition(olx_element.tag, olx_element.get('url_name'))
+        keys = ScopeIds(None, olx_element.tag, def_id, runtime.id_generator.create_usage(def_id))
+        block = WordCloudBlock.parse_xml(olx_element, runtime, keys)
```
As per discussion in pair programming session (@kdmccormick , @ttqureshi ) we are creating scope ids here which shouldn't happen and is a potential bug

kdmccormick replied:
Aha, here's why you needed to make this change! Passing in `None` for ScopeIds was a legacy behavior: https://github.com/openedx/edx-platform/blob/6d150acf81b07c72e85bc86e2171cffc3a7818a7/xmodule/xml_block.py#L309-L312

This change looks good 👍🏻

farhan commented:
@feanil You can consider this PR to be ready for next pass

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-8996948176.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-8996948176-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-8996948176-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-8996948176-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9036629849-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9036629849-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9036629849-tutor-requirements.txt)

kdmccormick commented:
(let's keep this marked as a Draft because it's not intended to be merged)

I've tried testing this a few ways, with mixed success. Let me know whether you think these might be bugs in the extracted block, or something I am doing wrong in my test environments.

### local xblock-sdk

I think it is working:
![image](https://github.com/user-attachments/assets/45d62639-7119-4454-ac24-b475df6c7d82)

I only see half the words I added, but maybe that's existing behavior.

### local edx-platform

Looks like the Studio editor works fine:
![image](https://github.com/user-attachments/assets/73d9a706-88b6-49a9-83b6-fc56e7c83cab)

But the LMS view shows nothing for me:
![image](https://github.com/user-attachments/assets/336bdd26-5846-494b-97b6-6023c7968f10)

### sandbox edx-platform

On https://app.pr-35983-139931.sandboxes.opencraft.hosting/learning/course/course-v1:Kyle+1+1/block-v1:Kyle+1+1+type@sequential+block@516bed191afe4262a0ee184df8a46167/block-v1:Kyle+1+1+type@vertical+block@142eaf5114464bdfa0b9983f16411cf5, (username/password = openedx/openedx), I get an error when I try to save words:

![image](https://github.com/user-attachments/assets/0d97c82e-b2a3-4f98-ae30-bc84d3befe1c)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547435.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547435-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547435-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547435-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547430-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547430-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9051547430-tutor-requirements.txt)

farhan commented:
@kdmccormick Thanks for doing the testing

> I only see half the words I added, but maybe that's existing behavior.

Yes, its existing behaviour. Often word cloud doesn't show all the words. You can refresh the page to get a refreshed experience.

> But the LMS view shows nothing for me:

It was an issue. I have updated the word cloud selector in this [commit](https://github.com/openedx/xblocks-contrib/pull/4/commits/6e70ecb1a117e54e67b9c15793c7f54fc59d6171) and redeploy the sandbox.
So everything should work good now 

Last thing looking into is https://github.com/openedx/edx-platform/pull/36199

cc: @ttqureshi @irtazaakram @feanil @salman2013

salman2013 commented:
On local edx-platform Studio works fine.

<img width="1415" alt="Screenshot 2025-02-06 at 3 34 20 PM" src="https://github.com/user-attachments/assets/56ac49d8-d3ae-4243-b737-e8446e67071f" />


When i preview it then it do not show anything, Please tell me if i do anything wrong or there is something need to configure, i am wondering as the video xblock is showing in preview mode.

<img width="1425" alt="Screenshot 2025-02-06 at 4 06 17 PM" src="https://github.com/user-attachments/assets/909c60f0-dfa0-405a-ae94-f8a746d7b6dd" />

@farhan

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9090220893-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9090220893-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9090220893-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9093926400.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9093926400-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9093926400-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9093926400-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9094578338.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9094578338-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9094578338-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9094578338-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9098255704-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9098255704-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9098255704-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9099996575-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9099996575-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9099996575-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9100436238.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9100436238-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9100436238-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9100436238-tutor-requirements.txt)

farhan commented:
@kdmccormick 
I think the settings configured in the PR description has effected the sandbox but it's not reflecting the changes accordingly.

For testing, I have turned the setting `USE_EXTRACTED_WORD_CLOUD_BLOCK` to false as it's already hard coded to true in this PR.

But this disabling is not reflecting on the sandbox environment with multiple formats applied.

To easily test the django settings flag we have added a data variable [here](https://github.com/openedx/xblocks-contrib/pull/4/commits/d167964989c55456b756f802e6d5937013594209)

I don't think so that it's something that is blocking the Word Cloud Extraction PR to be release.

cc: @openedx/axim-aximprovements , @ttqureshi

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9171763390-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9171763390-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9171763390-tutor-requirements.txt)

farhan committed:
chore: Fix test cases for extracted word cloud block

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181050561.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181050561-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181050561-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181050561-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181766715-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181766715-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9181766715-tutor-requirements.txt)

farhan commented:
@kdmccormick @irtazaakram @salman2013 
This PR is in merge state now.
Should we merge it or close it. Sooner or later we need it when we will make the extracted word cloud as a default one.

kdmccormick commented:
Let's keep this unmerged. I approved the xblocks-contrib PR without testing it much myself in order to unblock development of other blocks, so I'd like to take more time to test this before we make it the default.

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9185388408.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9185388408-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9185388408-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-9185388408-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10784465935.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10784465935-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10784465935-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10784465935-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556045.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556045-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556045-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556045-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556030-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556030-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10789556030-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10800705693.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10800705693-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10800705693-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10800705693-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10801428911-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10801428911-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10801428911-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10830666633.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10830666633-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10830666633-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10830666633-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10831800882.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10831800882-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10831800882-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10831800882-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10833676971-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10833676971-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10833676971-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10840798111-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10840798111-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10840798111-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10841109241-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10841109241-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10841109241-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10841457896-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10841457896-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10841457896-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335774.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335774-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335774-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335774-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335737-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335737-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10844335737-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10863318003-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10863318003-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10863318003-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304651.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304651-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304651-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304651-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304643-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304643-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10869304643-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872129551.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872129551-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872129551-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872129551-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872523687.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872523687-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872523687-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10872523687-tutor-requirements.txt)

ttqureshi reviewed (approved):
Looks good to me 👍 

I believe this documentation would help avoid any future confusions.

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10873289638.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10873289638-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10873289638-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10873289638-tutor-requirements.txt)

kdmccormick commented on xmodule/tests/test_word_cloud.py:
```diff
@@ -54,18 +61,41 @@ def test_xml_import_export_cycle(self):
         Make sure that answer for incorrect request is error json.
         """
+        if settings.USE_EXTRACTED_WORD_CLOUD_BLOCK:
+            return
```
This is essentially just skipping the test!

Could you run as much of the test as possible without the if, and then just use the if statement at the end for the final assertion?

kdmccormick commented on lms/djangoapps/courseware/tests/test_word_cloud.py:
```diff
@@ -18,6 +20,10 @@ class TestWordCloud(BaseTestXmodule):
     """Integration test for Word Cloud Block."""
```
We should ensure that these tests all run on CI for both the built-in block and the extracted block, regardless of the default setting value.

Here's a way to achieve this without having to change many lines:

The current test class is renamed to be just a base mixin (the "Mixin" suffix tells pytest not to execute the tests cases on this class):

```python
class _TestWordCloudMixin(BaseTestXmodule):
    ...
    ...  # all the test methods stay in this class
```

and then you have one concrete child class for the built-in block and one for the extracted block, both of which are empty (they'll inherit all their test cases from _TestWordCloudMixin):

```python
@override_settings(USE_EXTRACTED_WORD_CLOUD_BLOCK=False)
class TestBuiltinWordCloud(_TestWordCloudMixin):
    pass

@override_settings(USE_EXTRACTED_WORD_CLOUD_BLOCK=True)
class TestExtractedWordCloud(_TestWordCloudMixin):
    pass
```

How does that look? I think this should work both for this tests class and for the other WordCloud test class below, and I believe it will also work for other blocks like Poll once they're ready too.

kdmccormick reviewed (changes requested):
Thanks for the comments so far, they're helpful. This is getting close!

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10874873469.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10874873469-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10874873469-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10874873469-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10876547720.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10876547720-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10876547720-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10876547720-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10877296392-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10877296392-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10877296392-tutor-requirements.txt)

farhan committed:
chore: Address change requests

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10889832150.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10889832150-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10889832150-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10889832150-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893116254.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893116254-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893116254-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893116254-tutor-requirements.txt)

farhan commented:
@kdmccormick I have spent couple of days to run test cases both on built-in and extracted test cases.

You can find my existing code and debugging logs in last commit.

Let's take `lms/djangoapps/courseware/tests/test_word_cloud.py` as a sample

If i run tests enabling only 1 class (TestWordCloudExtracted or TestWordCloudBuiltIn) tests run successfully but when I enable both, one of them fails.

Based on my current assessment, it seems that the xmodule is caching the previously loaded Word Cloud block class (whether extracted or built-in), and it’s not being reset between test runs. This issue is still unresolved. Any thoughts, suggestions, or guidance would be greatly appreciated

cc: @feanil @openedx/axim-aximprovements

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893633258.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893633258-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893633258-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10893633258-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10900448541-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10900448541-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10900448541-tutor-requirements.txt)

farhan committed:
chore: Run test cases both for builtin and extracted blocks

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10904117043.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10904117043-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10904117043-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10904117043-tutor-requirements.txt)

farhan committed:
chore: update tests fix

farhan committed:
chore: fix pylint error

farhan commented:
@kdmccormick You can ignore last comment if didn't read yet. PR is available for the next pass.


_**Problem Cause:** So it was just `xblock.plugin.PLUGIN_CACHE` that was creating problem, reseting it on tests setUpClass._

farhan committed:
chore: fix quality check

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10906333569.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10906333569-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10906333569-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10906333569-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065960.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065960-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065960-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065960-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065876-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065876-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10907065876-tutor-requirements.txt)

ttqureshi commented:
From adding the docs against the `if` conditions to having the CI checks run for both the Built-in and the Extracted word cloud blocks, we've made a good progress in this PR.

But before we plan to merge this PR, I want to bring one point into consideration that is the [pointer tag](https://github.com/openedx/XBlock/issues/823) issue.

I think we can't merge this PR and any of the test PRs until we resolve this issue. Because the extracted blocks don't support the parsing of the pointer tags format properly which can lead to the erroneous parsing of the previously exported Builtin Blocks if they are imported in the pointer tag OLX format (see this [comment](https://github.com/openedx/XBlock/issues/823#issuecomment-2890952817)). For that, I am currently working on this issue.

@feanil @kdmccormick please share your thoughts on it.

cc to: @openedx/axim-aximprovements

farhan commented:
> I think we can't merge this PR and any of the test PRs until we resolve this issue.

@kdmccormick @ttqureshi 
I think in this case we should merge this PR keeping `USE_EXTRACTED_WORD_CLOUD_BLOCK` flag disabled for now, we can enable it default once the pointer tag issue implementation (discussed in last commit) is done.

kdmccormick commented:
Great work on the tests. Good point @ttqureshi , I will turn my attention back to the pointer tag issue. I agree with @farhan , we can merge this PR so tests run on the extracted block, but keep the feature flag disabled until pointer tags are fixed.

farhan committed:
chore: kept USE_EXTRACTED_WORD_CLOUD_BLOCK flag False for now

open-craft-grove commented:
Sandbox deployment failed 💥
Please check the settings and requirements.
Retry deployment by pushing a new commit or updating the requirements/settings in the pull request's description.
📜 [Failure Logs](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213508.log)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213508-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213508-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213508-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35983-139931.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35983-139931.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213505-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213505-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35983-139931-51564656-10938213505-tutor-requirements.txt)

kdmccormick reviewed (approved):
Great work!
